XdmodAID.md

```diff
@@ -0,0 +1,265 @@
+figured there was the distinct possiblity that some restructuring would be
+necessary.  However, I also wanted to see if the original docker files could be used
+as is, as this would potentally make upgrading easier as we would be using what
+their process developed.
+
+I had put some changes in to dockerfile to support xdmod-openstack and similar
+changes were done to support xdmod-shift.  Along the way, in order to get PRs
+approved, we decided that we needed to optimize the dockerfiles.
+
+Subsequently, I have moved the changes that I made to the original dockerfile
+to a different one based on python 3.11 dockerfile.  Likewise, the xdmod-openshift
+changes could be isolated.
+
+Currently, I am uncertain if the original dockerfile would still work for us, or if we
+could drop in the xdmod team's dockerfile.  I had planned to do this after
+moving xdmod-openshift into it's own dockerfile.
+
+#Basic Processing
+
+The overall processing flow is as follows
+```
+                  OpenStack                               OpenShift                 KeyCloak        ColdFront
+
+      xdmod-openstack   xdmod-openstack-hypervisors     xdmod-openshift                  xdmod-hierarchy
+
+      xdmod-shredder        xdmod-shredder              xdmod-shredder                   xdmod-import-csv
+
+
+                           xdmod-ingestor
+
+                                                       aggregate-supremm
+
+                                      acl-config
+
+                                      xdmod-ui
+
+                                batch_export_manager.php
+```
+
+
+#xdmod-openstack-hypervisors
+This script fetches the hypervisor information from openstack used in calculating
+percent usage.
+
+This was the simpliest script and required very few changes to work.
+
+There was a bug in the command line section of this script that is present in the
+upstream version, which we decided to fix with our version of the script. So our
+version will be different and possibly incompatible with the upstream version.
+
+This script produces a file specified on the command line that will need to be
+shredded.
+
+#acl-config
+This is a small utility that rebuilds the consistency between the configuration files
+and the database.  It is needed to be run as there are several processes that
+modify one or the other but not both.  It is an example of a work-a-round that
+became part of the processing.  From a conversation I had in a meeting with one of
+the xdmod developers, it is expected to eventually be eliminated, however it is
+currently low priority.
+
+As the xdmod-ui requires the database and configration files to be consistent,
+and we do not currently support RWX volumes, this needs to be run in parallel
+to the UI.
+
+Most of the time it runs withing a couple of minutes, sometimes it takes longer
+and will ocassionally fail due to a time out when it cannot lock certain tables.
+
+#xdmod-openstack
```

Comment:
did you mean to put anything in this section?

Reply:
I guess it is not obvious to the order that I am doing things.

I laid out sections that I want to eventually fill out.  The xdmod-openstack section is a bit larger and I started on it today.

Reply:
Im in the middle of the xdmod-openstack section - a lot more to write